client/browser/src/libs/bitbucket/scrape.ts

```diff
@@ -1,65 +1,94 @@
+ * or to the original source view, so a few different selectors are tried.
+ *
+ * The href of these links contains:
+ * - project name
+ * - repo name
+ * - file path
+ * - rev (through the query parameter 'at')
+ */
+const getFileInfoFromLink = (
+    codeView: HTMLElement
+): Pick<FileInfo, 'repoName' | 'filePath' | 'rev'> & BitbucketRepoInfo => {
+    const errors: Error[] = []
+    for (const selector of LINK_SELECTORS) {
+        try {
+            const linkElement = codeView.querySelector(selector) as HTMLLinkElement | null
+            if (!linkElement) {
+                throw new Error(`Could not find selector ${selector} in code view`)
+            }
+            const url = new URL(linkElement.href)
+            const host = window.location.hostname
+            const path = url.pathname
+
+            // Looks like /projects/<project>/repos/<repo>/(browse|raw)/<file path>?at=<rev>
+            const pathMatch = path.match(/\/projects\/(.*?)\/repos\/(.*?)\/(browse|raw)\/(.*)$/)
+            if (!pathMatch) {
+                throw new Error(`Path of link matching selector ${selector} did not match path regex: ${path}`)
+            }
+
+            const project = pathMatch[1]
+            const repoSlug = pathMatch[2]
+            const filePath = pathMatch[4]
+
+            // Looks like 'refs/heads/<rev>'
+            const at = url.searchParams.get('at')
+            if (!at) {
+                throw new Error(`href of link matching selector ${selector} did not have 'at' search param: ${url}`)
+            }
+
+            const atMatch = at.match(/refs\/heads\/(.*?)$/)
+
+            const rev = atMatch ? atMatch[1] : at
+
+            return {
+                repoName: [host, project, repoSlug].join('/'),
+                filePath,
+                rev,
+                project,
+                repoSlug,
+            }
+        } catch (err) {
+            errors.push(err)
+            continue
+        }
     }
+    throw new Error(`Could not parse file info from links in code view:\n${errors.map(e => `${e}`).join('\n')}`)
 }
 
-export const getFileInfoFromCodeView = (codeView: HTMLElement): PageInfo & Pick<FileInfo, 'commitID'> => {
-    const { repoName, filePath, rev, project, repoSlug } = getFileInfoFromLink(
-        codeView,
-        'a.raw-view-link',
-        // Looks like '/projects/<project>/repos/<repo name>/raw/<file path>'
-        /\/projects\/(.*?)\/repos\/(.*?)\/raw\/(.*)$/
-    )
-
+/**
+ * Attempts to retreive the commitid from a link to the commit,
+ * found on single file "diff to previous" views.
+ */
+const getCommitIDFromLink = (): string => {
     const commitLink = document.querySelector<HTMLElement>('a.commitid')
     if (!commitLink) {
-        throw new Error('Could not find commit id link')
+        throw new Error('No element found matching a.commitid')
     }
-
     const commitID = commitLink.dataset.commitid!
```

Comment:
```suggestion
     const commitID = commitLink.dataset.commitid
```
since you are checking it below